### PR TITLE
Use pnpm as package manager

### PR DESCRIPTION
**Pull request by nsams**
additionally:

- fix all peerDependency issues (that where present before too)
- update all packages (as new lock file was created)
- make scripts package manager independent
- don't rely on dependencies in root package.json anymore
- fix several issues related to libraries updates (see individual commits)
- convert peerDependencies to dependencies where it makes sense
- NEW: add changesets for tracking changelogs (and releasing)

Why?

- pnpm is even faster than yarn
- it creates a non-flat node_modules, that is so much better to work with as it includes only directly used packages
- we don't need lerna anymore
- more flexible regarding peerDependencies (configurable autoinstall)

TODO

- [x] convert scripts currently using lerna
- [x] fix storybook build
- [x] fix demo build
- [x] experiment with wml/link/whatever setup: done, nothing that could replace wml found
- [x] release workflow (https://pnpm.io/using-changesets?)
- [x] fix deploy to netlify

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -23,38 +23,52 @@ jobs:
...
             - name: Build
-              run: yarn build:lib
+              run: pnpm run build:lib
 
             - name: Lint
               run: |
-                  yarn exec prettier --check "./*.{js,json,md,yml}"
-                  yarn lint:lib
+                  pnpm exec prettier --check "./*.{js,json,md,yml}"
+                  pnpm run lint
```
Comment (nsams) [inline .github/workflows/ci.yml:66]: demo was not linted before(?)

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -64,46 +78,30 @@ jobs:
                   path: packages/**/junit.xml
                   reporter: jest-junit
 
-            - name: "Library: Clone translations"
```
Comment (nsams) [inline .github/workflows/ci.yml:67]: moved up as linting demo required translations

**Comment by kaufmo** [comment]
Some questions from my side as I dont know what are your thoughts behind this MR why we need this?
1. as we are already using yarn berry why not using Yarn PnP?
2. It looks to me that this has taken much time to switch to pnpm is this worth?

Two questions you wanted answered from me:
> Should we use pnpm in Comet?  

For me it is OK when we say we use it for monorepos. I see the benefits but are they so big compared to Yarn PnP or NPM v9 for example? 

> Should we use pnpm for projects to?  

No I actually think not, I think the main benefits are for monorepos and as we develop only singlerepos I see no reason to switch here

File: .github/workflows/release-canary.yml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            - name: Publish Canary release
+              if: ${{ github.ref_name == 'next' }}
+              run: |
+                  echo --- > .changeset/canary.md
+                  echo '"@comet/cli": major' >> .changeset/canary.md
+                  echo --- >> .changeset/canary.md
+                  echo >> .changeset/canary.md
+                  echo fake change to always get a canary release >> .changeset/canary.md
+                  pnpm exec changeset version --snapshot canary
+                  pnpm run publish --tag next-canary
```
[the remark this answers is not in the document]
Comment (nsams) [reply .github/workflows/release-canary.yml:55]: > Or should we reserve next for possible alpha/beta releases of next?

yes, that was my thought behind this

File: packages/admin/blocks-admin/src/index.ts [hunk 4]
```diff
@@ -81,3 +81,7 @@ export { SelectPreviewComponent } from "./iframebridge/SelectPreviewComponent";
 export { useIFrameBridge } from "./iframebridge/useIFrameBridge";
 export { parallelAsyncEvery } from "./utils/parallelAsyncEvery";
 export { isValidUrl } from "./validators/isValidUrl";
+// eslint-disable-next-line @typescript-eslint/no-unused-vars, unused-imports/no-unused-imports
+import emotionStyled from "@emotion/styled";
```
[the remark this answers is not in the document]
Comment (nsams) [reply packages/admin/blocks-admin/src/index.ts:85]: Not really, sorry.

"I don't fully understand the issue or the solution."

**Review by johnnyomair — APPROVED** [review]

File: packages/admin/blocks-admin/src/index.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (johnnyomair) [reply packages/admin/blocks-admin/src/index.ts:85]: I did some research on my own. It seems that this is related with PNPM's .pnpm/ folder (See the following issue: https://github.com/microsoft/TypeScript/issues/42873).

**Review by johnnyomair — DISMISSED** [review]

File: install.sh [hunk 5]
```diff
@@ -7,8 +7,8 @@ cd $(dirname $0)
 # use correct node version and install dependencies
 nvm install
 nvm use
-npm i -g yarn
-yarn install
+npm i -g pnpm@7
```
Comment (dkarnutsch) [inline install.sh:10]: PNPMv8 was released two days ago. Might it be worth it, to directly target that version? (see https://github.com/pnpm/pnpm/releases)

File: packages/admin/cms-admin/package.json [hunk 6]
```diff
@@ -1,51 +1,47 @@
...
-        "build": "yarn run clean && run-p generate-graphql-types generate-block-types && run-p build:babel build:types",
+        "build": "$npm_execpath run clean && run-p generate-graphql-types generate-block-types && run-p build:babel build:types",
         "build:babel": "npx babel ./src -x \".ts,.tsx\" -d lib",
         "build:types": "tsc --project ./tsconfig.json --emitDeclarationOnly",
-        "start": "run-p generate-graphql-types generate-block-types && run-p start:babel start:types",
-        "start:babel": "npx babel ./src -x \".ts,.tsx\" -d lib -w",
-        "start:types": "tsc --project ./tsconfig.json --emitDeclarationOnly --watch --preserveWatchOutput",
+        "clean": "rimraf lib",
+        "generate-block-types": "comet generate-block-types --inputs",
+        "generate-block-types:watch": "chokidar -s \"**/block-meta.json\" -c \"$npm_execpath generate-block-types\"",
```
Comment (fraxachun) [inline packages/admin/cms-admin/package.json:21]: chokidar-cli is missing as peerDependency

**Review by fraxachun — DISMISSED** [review]

**Comment by dkarnutsch** [comment]
@nsams what is the timeline for this PR? Should we use PNP8 directly? (https://github.com/vivid-planet/comet/pull/974#discussion_r1152837225)

I would like to update @nestjs/graphql in order to update to apollo-server v4, but I'm awaiting this change.

File: install.sh (hunk 5 above)
Comment (nsams) [reply install.sh]: done

**Review by johnnyomair — APPROVED** [review]

**Review by dkarnutsch — APPROVED** [review]

**Pull request merged**